src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java

```diff
@@ -0,0 +1,20 @@
+package com.wolox.training.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class RestExceptionHandler {
+
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity notFoundException (Exception e){
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
```

Comment:
I don't know if this is already happening (i can't download the code right now 😋), but you should inform the user that the thing that was not found was a `Book` and not a String for example. A message "book not found" or similar should appear somewhere. If this is already happening, please ignore this comment.

Reply:
Fixed in https://github.com/wolox-training/gf-java/pull/9